### DISCUSSION TITLE
fslreorient2std functionality in ANTs?

### DISCUSSION THREAD

**Discussion by lidialuq** [ANTsX/ANTs#1288] [Q&A]
I find ANTs superior to FSL in many ways, but I haven't yet found a function that replaces fslreorient2std in ANTs. Here's what is does, from the docs: 

" This is a simple tool designed to reorient an image to match the orientation of the standard template images (MNI152) so that they appear "the same way around" in FSLView. It requires that the image labels are correct in FSLView before this is run. It is also not a registration tool, so it will not align the image to standard space, it will only apply 90, 180 or 270 degree rotations about the different axes as necessary to get the labels in the same position as the standard template. "

It is possible to archieve the same with ANTs?

**Comment by gdevenyi** [comment]
This might do what you want.
```
PermuteFlipImageOrientationAxes 
Usage: 
PermuteFlipImageOrientationAxes ImageDimension  inputImageFile  outputImageFile xperm yperm {zperm}  xflip yflip {zflip}  {FlipAboutOrigin}
 for 3D:  PermuteFlipImageOrientationAxes 3  in.nii out.nii   2 0 1  1 1 1  
 would map z=>x, x=>y, y=>z and flip each 
 for 2D:  PermuteFlipImageOrientationAxes 2  in.nii out.nii   1 0  1 0  
 would map x=>y, y=>x and flip x  

 0 1 2 for permute factors gives no axis permutation 
 1 2 0 maps y to x,  z to y and x to z 
 the flip values are boolean -  0 1 0 would flip the y-axis only 

 The FlipAboutOrigin boolean lets you flip about the coordinate set in the origin
```

However ANTs lives in ITK-land and itksnap is the viewer that is used to determine if loading/viewing is correct.

**Comment by tangYING2022** [comment]
Excuse me, a problem occurred to me when l was trying to do coregistration about orientation, 
how can l ensure the images waiting for coregistration are in the same orientation？ 
Whether the images opened in different windows of ITK-SNAP don't appear mirrored means they are in the same orientation, although the ijk coordinate system are different.
    l would be appreciated if you can offer me some suggestions.

**Reply by cookpa** [reply]
If the anatomical labels in ITK-SNAP agree, and the crosshairs are roughly in the same place, you're probably good to go.

ANTs will start with a center of mass transform. You can preview this using ITK-SNAP's registration tool. If you like, you can also manually initialize registration, save the transform as an ITK transform, and pass it to `antsRegistration` with `-r`.

**Reply by tangYING2022** [reply]
Thanks for your kind help.
